constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java

```diff
@@ -1801,11 +1801,17 @@ protected HttpUrl makeUrl(List<String> paths, UserInfo info)
         okhttp3.HttpUrl.Builder builder =
                 new HttpUrl.Builder()
                         .scheme(this.protocol)
+                        .host(this.host)
                         .addQueryParameter("key", this.apiKey)
-                        .addQueryParameter("c", this.version)
-                        .addQueryParameter("s", String.valueOf(info.getSessionId()))
-                        .addQueryParameter("i", info.getClientId())
-                        .host(this.host);
+                        .addQueryParameter("c", this.version);
+
+        if (info.getClientId() != null) {
+            builder.addQueryParameter("i", info.getClientId());
+        }
+
+        if (info.getSessionId() != 0) {
```

Comment:
Does session id default to 0 (since it's an int) or should we also check for null here?

Reply:
I tested it and ints default to zero and Integer defaults to null so we're fine here.

Reply:
Cool beans. 🚀